### PR TITLE
Add support for Audio Tokenization and Voice Emulation via SNAC Tokenization

### PR DESCRIPTION
**Pull request by klei22**
This PR creates a new folder in the data directory for snac processing.

See README.md for 3 step demo how to emulate voice timbre based on a collection of mp3 files.

Current settings are created for voice, but we can combine this with other methods for further reducing context length.

### CODE REVIEW

File: data/snac/get_dataset.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+out_dir="input_mp3s"
+txt_dir="texts"
+
+if [[ ! -d "${out_dir}" ]]; then
+  mkdir -p "${out_dir}"
+fi
+
+for i in $(seq -f "%02g" 1 24); do
+  wget -nc -O "${out_dir}/tiny_sherlock_audio_${i}.mp3" "${url}/resolve/main/adventuresholmes_${i}_doyle_64kb.mp3?download=true"
+donej
```
Comment (gkielian) [inline data/snac/get_dataset.sh:22]: Just tried this, there seems to be an extra j here

File: data/snac/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+```
+4000 300 400 133 3000 1234 1337 `4097` 100 2000 3000 3999 4000 4001 4002 `4097`
+```
+
+Each of these numbers counts as a single token, and spaces are skipped for
+maximum context length capability.
+
+## Usage
+
```
Comment (gkielian) [inline data/snac/README.md:37]: Add section for dependences

```
pip install snac
```

```
sudo apt update
sudo apt install ffmpeg
```

**Review by gkielian — CHANGES_REQUESTED** [review]
Looks great! : D

Adding preliminary review, currently going through each the steps in the README.md.

File: data/snac/get_dataset.sh (hunk 1 above)
Comment (klei22) [reply data/snac/get_dataset.sh]: done

**Review by gkielian — APPROVED** [review]

**Pull request merged**